deps/spidershim/src/v8object.cc

```diff
@@ -454,6 +454,38 @@ Local<Object> Object::Clone() {
   JSContext* cx = JSContextFromIsolate(isolate);
   AutoJSAPI jsAPI(cx, this);
   JS::RootedObject thisObj(cx, GetObject(this));
+  bool isArray;
+  if (!JS_IsArrayObject(cx, thisObj, &isArray)) {
+    return Local<Object>();
+  }
+
+  if (isArray) {
+    uint32_t length;
+    if (!JS_GetArrayLength(cx, thisObj, &length)) {
+      return Local<Object>();
+    }
+
+    JS::RootedObject copy(cx, JS_NewArrayObject(cx, length));
+    if (!copy) {
+      return Local<Object>();
+    }
+
+    JS::RootedValue val(cx);
+    for (uint32_t i = 0; i < length; i++) {
+      if (!JS_GetElement(cx, thisObj, i, &val)) {
```

Comment:
So... this won't do the "right thing" for accessor properties.  For that matter, it won't copy non-integer-named properties.  Or preserve holes.  I'm not sure what semantics you want here, though.